pkg/migration/generate.go

```diff
@@ -384,6 +420,19 @@ func (h *Handler) constructStorageCluster(ds *appsv1.DaemonSet) *corev1.StorageC
 	return cluster
 }
 
+func (h *Handler) removeCustomImageRegistry(image, customImageRegistry string) string {
+	if image == "" || customImageRegistry == "" {
+		return image
+	}
+
+	if !strings.HasPrefix(image, customImageRegistry) {
+		logrus.Warningf("image %s does not have custom image registry prefix %s", image, customImageRegistry)
+		return image
+	}
+
+	return strings.TrimPrefix(image, customImageRegistry+"/")
```

Comment:
What would happen if the custom registry is `registry.io/custom-namespace`? 
The image in this case would be `registry.io/custom-namespace/stork:x.x.x`. If we trim it, then would just `stork:x.x.x` remain? It might still work, but just verify that it works with just the image portion.

Reply:
yes, that is expected. as we would add it back in getImageURN

Reply:
I mean let's verify that we don't end up using default and that the component deployments come up fine even with just name in the `status.desiredImages`.

Reply:
I will add automated functional test for it after FC.

Reply:
I just wanted to know if we verified this case manually at least, where the image in desired images is just the name (eg. `stork:2.7.1`) and no path